src/workflows/airqo_etl_utils/bigquery_api.py

```
@@ -846,6 +845,79 @@
         except Exception as e:
             print("Error fetching data from bigquery", {e})
 
+    def fetch_device_data_for_satellite_job(
+        self,
+        start_date_time: str,
+        job_type: str,
+    ) -> pd.DataFrame:
+        try:
+            pd.to_datetime(start_date_time)
+        except ValueError:
+            raise ValueError(f"Invalid start date time: {start_date_time}")
```

Ruff

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)